.evergreen/config.yml

```diff
@@ -974,18 +974,26 @@ functions:
           ${PREPARE_SHELL}
 
           cd ${DRIVERS_TOOLS}/.evergreen/csfle
-          source activate_venv.sh || :
+
+          # only run this if virtualenv is installed
+          if ${PYTHON3_BINARY} -m venv ./venv; then
```

Comment:
Were we getting errors from virtualenv not being installed? Or is this just precautionary?

Reply:
Yes, we get this error on [3.0 Ubuntu 14.04](https://evergreen.mongodb.com/task/mongo_go_driver_tests_legacy_auth_ssl__version~3.0_os_ssl_legacy~ubuntu1404_go_1_16_test_replicaset_auth_nossl_patch_404a99a20576ef7d86f62910917c53baf0dd9319_62bdfa28d1fe072a6ff02a25_22_06_30_19_32_17##comparehashes=404a99a20576ef7d86f62910917c53baf0dd9319&threads=maxonly):

```
[2022/06/30 19:38:34.921] /usr/bin/python3: No module named virtualenv
```

Reply:
Should we not even try to start CSE servers when virtualenv is not available, then? Otherwise, I imagine 3.0 Ubuntu will start system failing.

Reply:
Ah I see, I keep forgetting that because we have `pre_error_fails_tasks: true` in the config, using something like `exit 1` in a `pre` task will cause the entire test to fail. I think the current setup is fine, then. 

To check my understanding: if `virtualenv` is not an available module attached to the `PYTHON3_BINARY`, we'll echo `"error: no Module named venv"`, and the background task that starts the CSE servers in the next `command` will fail silently?

Reply:
That is my understanding, here is a test that does that: https://evergreen.mongodb.com/task/mongo_go_driver_tests_legacy_noauth_nossl__version~3.0_os_ssl_32~ubuntu1604_64_go_1_17_test_replicaset_noauth_nossl_patch_404a99a20576ef7d86f62910917c53baf0dd9319_62be16763e8e866769a62b83_22_06_30_21_33_07##comparehashes=404a99a20576ef7d86f62910917c53baf0dd9319&threads=maxonly

We could wrap the logic that starts the server in the next command, though? IIUC the background job will error silently, but that may not be the most graceful way of handling this?